### PR TITLE
#515 Optimise dependencies to remove polyfil

### LINKED ISSUE #515: eliminate explicit dependency for STJ on newer frameworks
**Is your feature request related to a problem? Please describe.**
I want to minimise dependencies in my project by utilising framework dependencies wherever possible

**Describe the solution you'd like**
I want the package to have conditions placed on System.Text.Json so that it can be provided by the framework. Same thing for System.ComponentModel.Annotations

**Describe alternatives you've considered**
Accept the additional dependency

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by thompson-tomo**
Have also enabled global package reference

Closes #515

### CODE REVIEW

**Review by fiseni — APPROVED** [review]

File: src/SmartEnum/SmartFlagEnum.cs [hunk 1]
```diff
@@ -300,15 +299,16 @@ public static bool TryFromValueToString(TValue value, out string result)
...
         }
 
+        [SuppressMessage("Minor Code Smell", "S6608:Prefer indexing instead of \"Enumerable\" methods on types implementing \"IList\"", Justification = "<Pending>")]
         private static string FormatEnumListString(IEnumerable<TEnum> enumInputList)
         {
             var enumList = enumInputList.ToList();
             var sb = new StringBuilder();
 
-            foreach (var smartFlagEnum in enumList)
+            foreach (var smartFlagEnum in enumList.Select(x => x.Name))
```
[the remark this answers is not in the document]
Comment (thompson-tomo) [reply src/SmartEnum/SmartFlagEnum.cs:308]: This was a warning/error being thrown by the compiler for the newer TFM. This change allows the simplification below.

File: src/SmartEnum/SmartFlagEnum.cs [hunk 2]
```diff
@@ -199,24 +201,21 @@ public static IEnumerable<TEnum> FromValue(TValue value)
...
-        /// <exception cref="SmartEnumNotFoundException"><paramref name="value"/> does not exist.</exception> 
+        /// <exception cref="SmartEnumNotFoundException"><paramref name="value"/> does not exist.</exception>
         /// <param name="value"></param>
         /// <returns></returns>
         public static TEnum DeserializeValue(TValue value)
         {
             var enumList = GetAllOptions();
 
-            foreach (var smartFlagEnum in enumList)
+            var returnValue = enumList.FirstOrDefault(x => x.Value.Equals(value));
```
[the remark this answers is not in the document]
Comment (thompson-tomo) [reply src/SmartEnum/SmartFlagEnum.cs:211]: Done.

File: src/SmartEnum.ModelBinding/SmartEnumModelBinder.cs [hunk 3]
```diff
@@ -36,14 +36,11 @@ public Task BindModelAsync(ModelBindingContext bindingContext)
                 if (methodInfo.Name == "FromName")
                 {
                     ParameterInfo[] methodsParams = methodInfo.GetParameters();
-                    if (methodsParams.Length == 2)
+                    if (methodsParams.Length == 2 && methodsParams[0].ParameterType == typeof(string) && methodsParams[1].ParameterType == typeof(bool))
```
[the remark this answers is not in the document]
Comment (thompson-tomo) [reply src/SmartEnum.ModelBinding/SmartEnumModelBinder.cs]: Done.

**Review by fiseni — APPROVED** [review]

**Comment by fiseni** [comment]
Thanks. @ardalis this is ready for merging.

**Comment by ardalis** [comment]
Done

**Pull request merged**